cowprotocol/accounting/rewards/mainnet/mainnet_dashboard_query_2510345.sql

```diff
@@ -0,0 +1,206 @@
+                and date(minute) = cast('{{end_time}}' as timestamp) - interval '1' day
+        ) as cow_price,
+        (
+            select avg(price) from prices.usd
+            where
+                blockchain = 'ethereum'
+                and contract_address = 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
+                and date(minute) = cast('{{end_time}}' as timestamp) - interval '1' day
+        ) as eth_price
+),
+
+-- BEGIN QUOTE REWARDS
+order_quotes as (
+    select
+        order_uid,
+        quote_solver
+    from cow_protocol_ethereum.order_rewards
+    where
+        block_number > (select start_block from block_range)
+        and block_number <= (select end_block from block_range)
+),
+
+winning_quotes as (
+    select
+        oq.order_uid,
+        quote_solver as solver
+    from order_quotes as oq
+    inner join cow_protocol_ethereum.trades as t on oq.order_uid = t.order_uid and oq.quote_solver != 0x0000000000000000000000000000000000000000
+),
+
+quote_rewards as (
+    select
+        solver,
+        least({{quote_reward}}, {{quote_cap_eth}} * (select eth_price / cow_price from conversion_prices)) * count(*) as quote_reward
+    from winning_quotes group by solver
+),
+
+aggregate_results as (
+    select
+        pr.solver,
+        coalesce(reward_wei, 0) / pow(10, 18) as primary_reward_eth,
+        coalesce(network_fee_wei, 0) / pow(10, 18) as network_fee_eth,
+        coalesce(execution_cost_wei, 0) / pow(10, 18) as execution_cost_eth,
+        coalesce(reward_wei, 0) / pow(10, 18) * (select eth_price / cow_price from conversion_prices) as primary_reward_cow
+    from primary_rewards as pr left outer join fees_and_costs as fc on pr.solver = fc.solver
+),
+
+combined_data as (
+    select
+        coalesce(ar.solver, ss.solver, qr.solver) as solver,
+        network_fee_eth,
+        execution_cost_eth,
+        primary_reward_eth,
+        primary_reward_cow,
+        coalesce(quote_reward, 0) as quote_reward,
+        coalesce(slippage, 0) as slippage_eth,
+        concat(
+            '<a href="https://dune.com/queries/2332678?SolverAddress=',
+            cast(ar.solver as varchar),
+            '&start_time={{start_time}}&end_time={{end_time}}&min_absolute_slippage_tolerance=0&relative_slippage_tolerance=0&significant_slippage_value=0" target="_blank">link</a>'
+        ) as slippage_per_tx,
+        concat(environment, '-', name) as name  --noqa: RF04
+    from aggregate_results as ar
+    full outer join solver_slippage as ss
+        on ar.solver = ss.solver
+    full outer join quote_rewards as qr
+        on ar.solver = qr.solver
+    left join cow_protocol_ethereum.solvers as s
+        on coalesce(ar.solver, ss.solver, qr.solver) = s.address
+),
+
+service_fee_flag as (
+    select
+        solver,
+        service_fee,
+        case
+            when service_fee is true then 0.85
+            else 1
+        end as service_fee_factor
+    from "query_4017925"
```

Comment:
This will need to be passed a `start_time` parameter since we need the status on the first day of the accounting period.

Reply:
Oh yeah, technically you are right here. The payouts script doesn't need that but we might run into edge cases (i.e., one week for each solver) where this might incorrectly charge a fee as it currently checks roughly the final day of the accounting and not the first day.

Reply:
In any case, the service fee query requires fixing but it seems its results are fine for today so i would leave it as is in order not to get stuck at this

Reply:
#38